Adding installation instructions for Windows

llrs opened the pull request:
This PR adds the instructions for windows, both binaries and from source.
Lately I cannot configure R from source in the windows but following this should be possible.

If someone could confirm or report any problem on this instructions I would appreciate and update the instructions. 

----

This should be rebased and merged after #65

hturner commented:
I tried out the instructions. When I get to `$TOP_SRCDIR/configure --enable-R-shlib` I get the error `line 24073: ${}: bad substitution` which corresponds to the line `eval "test -z \"\${${Rshlibpath_var}}\""` in the configure script. 

I think this error is due to `Rshlibpath_var` being empty, which you can confirm by running

    Rshlibpath_var=test
    eval "test -z \"\${${Rshlibpath_var}}\""

in the MYS2 shell - this works as above, but if you replace `test` with `""` , you get the `${}: bad substitution` error.

The wider problem is that when running `$TOP_SRCDIR/configure` (with or without the `--enable-R-shlib` option) from the MYS2 shell, the host system type is identified as x86_64-pc-msys and looking at the configure file, the variable shlibpath_var is not set in this case.  If I run the same code in the Git Bash terminal in RStudio on Windows, the host system type is identified as x86_64-pc-mingw64 and the configure script gets past line 24073.

However, that doesn't solve the problem. The configure script then gets stuck because it can't find readline; if I run with option `--with-readline=no` it then gets stuck because it can't find X11; if I add option `--with-x=no` it gets stuck at `checking for BSD networking... configure: BSD networking functions are required`. I couldn't find a way to get past this!

Looking again at https://cran.r-project.org/bin/windows/base/howto-R-4.2.html, those instructions do not use the configure script on Windows. I was able to build R successfully by following the instructions there to build directly in the `src/gnuwin32` sub-directory of the svn repo. I also ran make check and make check-devel - the only issues were related to package vignettes and manuals, which are built in separate make steps (this isn't documented, you have to look in the `Makefile` in `src/gnuwin32`). I didn't get as far as fixing all that as I don't have MikTeX installed (I had TinyTex, but that's not easy to use as you have to install all the required packages manually when using outside of R).

So I think the bottom line is we shouldn't be using the configure script for Windows. We don't need to use the `--enable-R-shlib` to use the built version of R with RStudio, you can just go to Tools > Global Options > R version: [Change] > Choose a specific version of R and browse to `~/Downloads/R` (i.e. `$TOPSRC_DIR` in the instructions). I was able to get this work and ran the examples in `?plots` to check.

hturner commented on 02-getting_started.Rmd:
```diff
@@ -1,67 +1,249 @@
+    You will be prompted to select components at this stage: `User installation`, `64-bit User installation`, or `Custom installation`.
+    The default option may be chosen for the questions from this step onwards to complete the installation.
+    
+You can also use the [binaries for `r-devel` made available](https://cran.r-project.org/bin/windows/base/rdevel.html) which are not so up to date. 
```
I would not be so negative here, they are daily builds, so may be in sync with Subversion or a small number of  commits behind, which will rarely affect people's work.

hturner commented on 02-getting_started.Rmd:
```diff
@@ -1,67 +1,249 @@
+1.  Open the Rtools42 terminal, and update and get the tools:
+
+      pacman -Syuu
+      pacman -Sy wget subversion
```
We probably don't need subversion here if we are using TortoiseSVN or SlikSVN. I think it is helpful to recommend people install Subversion outside of the MSYS shell, so that it can be found more easily by other tools, e.g. RStudio.

hturner commented on 02-getting_started.Rmd:
```diff
@@ -1,67 +1,249 @@
+      
+2. Download the latest tcl/tk
+
+      wget -np -nd -r -l1 -A 'tcltk-*.zip' https://cran.r-project.org/bin/windows/Rtools/rtools42/files/
```
You currently just download the .zip file. It needs to be unpacked in $TOPSRC_DIR.

hturner commented on 02-getting_started.Rmd:
```diff
@@ -1,67 +1,249 @@
+      
+3. Set environment variables for POSIX, MiKTeX, tar (and one tar option):
+
+        export PATH=/x86_64-w64-mingw32.static.posix/bin:$PATH
```
This line is adding the path to the toolchain and libraries (they are inside /x86_64-w64-mingw32.static.posix which corresponds to c:\rtools42\x86_64-w64-mingw32.static.posix outside the shell). This ensures that gcc can be found.

hturner commented on 02-getting_started.Rmd:
```diff
@@ -1,67 +1,249 @@
+    If there is any error you'll need to find where the program is installed and add the corresponding path.
+1.  Download the latest recommended packages[^02-getting_started-2]:
+
+        $TOP_SRCDIR/tools/rsync-recommended
```
I tested the instructions on a user account named "Test account" - this caused issues with the use of variables because of the space in the username. This can be fixed by using quotes.
```suggestion
        "$TOP_SRCDIR/tools/rsync-recommended"
```
This applies to all uses of `$TOP_SRCDIR` and `$BUILDDIR`.

llrs commented:
Many thanks as always for helping me solve this problems! 

I'll try to build it on Sunday, but when you said you were able to " to build R successfully " do you mean just running `make` on the `src/gnuwin32`?  
I thought that before that I needed to configure it (as well as for RStudio to be able to use it). 

I have MikTeX installed on that computer, and LiveTeX which provides the `pdflatex` command, I'll report if I can build the manuals and update how it goes.

hturner commented:
Not quite, the make has different targets (`rsync-recommended`, `all` and `recommended`). So I did

    cd ~/Downloads/R
    unzip ../tcltk-5355-5175.zip
    cd src/gnuwin32
    make rsync-recommended
    make all recommended
    make check
    make check-devel
    
I think `make vignettes manuals` might be needed for the checks to pass without any warnings or notes, but I guess this is also true for the Linux installation, so perhaps we can note that as an optional step.

llrs commented:
Just notes for myself before writing it and anyone following through.

I can confirm this src/gnuwin32 works. But so far I used texlive as the pdflatex provider (in /c/texlive/2022/bin/win32/). When I opened the miketex, via the windows start menu, it asked me to check for updates and make it available in the PATH. I think it might be worth it to mention that one needs to open it and follow the recommendations then it would find the pdflatex file.

Also, I was surprised because RStudio wanted to select a folder with the R installation not the R.exe as I expected. This might be me not reading well the butttons in addition of lack of fmailiariy with windows programs.

llrs commented:
I was able to build and check successfully R from source in windows with MiKTeX, the program was installed just for the user in /c/Users/$USER/AppData/Local/Programs/MiKTeX/miktex/bin/x64.

I still need to check if `make check-devel` has any problem using user installation of MiKTeX, but I don't expect any. Another question is why with texlive the check fails...

llrs commented:
I updated the branch to (hopefully) resolve all the comments. I kept the svn installation inside MSYS to keep in line with other installation instructions but I mention there SlikSVN and TortoiseSVN.

nbenn reviewed (commented):
Unfortunately I neither have a Win machine available, nor have I ever tried building anything on that platform, so I can't give much feedback on content. I tried to improve some language issues, but feel free to ignore comments you don't see as improvements.

llrs commented on 02-getting_started.Rmd:
````diff
@@ -27,190 +27,272 @@ The `r-devel` at [R/trunk](https://svn.r-project.org/R/trunk "svn devel source c
+    svn update
+    ```
+
+    You can also use a SVN client such as TortoiseSVN (<https://tortoisesvn.net/>, command line tool, and Windows Explorer integration) or SlikSVN (<https://sliksvn.com/download/>, just the command line tool) so that it can be also found by other tools.
````
This comes from a previous comment from Heather to install subversion so that Rstudio or any other developer tool can use them. I only used TortoiseSVN and it worked via right click menu in Windows.

llrs commented on 02-getting_started.Rmd:
```diff
@@ -27,190 +27,272 @@ The `r-devel` at [R/trunk](https://svn.r-project.org/R/trunk "svn devel source c
 
-For development and testing, you need only the quick development build.
-The quick build avoids building the manuals, which are generally not needed for development and testing.
+6.  Build R and the recommended files:
```
Right it should be recommended packages.

llrs commented:
I think now I addressed all the comments, if there isn't any other big issue/comment I will merge this after the meeting tomorrow. Thanks for all the feedback @hturner and @nbenn !

hturner commented:
No major comments from me, so I merged this in to enable people to review the new section in a clean state.
